File: javascript/robot-name/robot_name.js
var generatedNames = [];
function generateName() {
  var name = [
    String.fromCharCode(65 + Math.floor(Math.random() * 25)),
    String.fromCharCode(65 + Math.floor(Math.random() * 25)),
    Math.floor((Math.random() * 10)),
    Math.floor((Math.random() * 10)),
    Math.floor((Math.random() * 10))
  ].join('');
  if (generatedNames.indexOf(name) > -1) {
    return generateName();
  }
  generatedNames.push(name);
  return name;
}

var Robot = function() {
  this.name = generateName();
};

Robot.prototype.reset = function() {
  this.name = generateName();
};

module.exports = Robot;

File: python/robot-name/robot_name.py
from string import ascii_uppercase
from random import sample, shuffle

usedNames = []
def generateName():
    alphabet = list(ascii_uppercase)
    letters = sample(alphabet, 2)
    digits = sample(range(9), 3)
    name = ''.join(str(x) for x in letters + digits)
    if name not in usedNames:
        usedNames.append(name)
        return name
    else:
        return generateName()

class Robot(object):
    def __init__(self):
        self.name = generateName()

    def reset(self):
        self.name = generateName()

